code/perception/src/radar_node.py

```diff
@@ -28,46 +111,157 @@ def callback(self, data):
-            (dataarray, clustered_data.labels_.reshape(-1, 1))
+            (combined_points, clustered_data.labels_.reshape(-1, 1))
         )
         bounding_boxes = generate_bounding_boxes(points_with_labels)
 
         marker_array = MarkerArray()
         for label, bbox in bounding_boxes:
-            if label != -1:
-                marker = create_bounding_box_marker(label, bbox)
-                marker_array.markers.append(marker)
-                # can be used for extra debugging
-                # min_marker, max_marker = create_min_max_markers(label, bbox)
-                # marker_array.markers.append(min_marker)
-                # marker_array.markers.append(max_marker)
+            rospy.loginfo(f"Label: {label}, Bounding Box: {bbox}")
+            marker = create_bounding_box_marker(
+                label, bbox, bbox_lifetime=self.data_buffer_time
+            )
+            # marker = create_moving_bbox_marker(label, bbox)
+            marker_array.markers.append(marker)
 
+        rospy.loginfo(f"Publishing {len(marker_array.markers)} markers.")
         self.marker_visualization_radar_publisher.publish(marker_array)
 
         cluster_info = generate_cluster_info(
-            clustered_data, dataarray, marker_array, bounding_boxes
+            clustered_data, combined_points, marker_array, bounding_boxes
         )
+        rospy.loginfo(f"Cluster Info: {cluster_info}")
         self.cluster_info_radar_publisher.publish(cluster_info)
 
+        # Setze die gespeicherten Nachrichten zurück
+        self.sensor_data = {key: None for key in self.sensor_data}
+
+    def extract_points(self, msg, sensor_name):
+        """
+        Extracts and transforms points from a ROS PointCloud2 message based on
+          a specified sensor configuration.
+
+        Parameters:
+        - msg: sensor_msgs/PointCloud2
+            The ROS PointCloud2 message containing the 3D points.
+        - sensor_name: str
+            The name of the sensor, used to retrieve its transformation parameters from
+            the sensor configuration.
+
+        Returns:
+        - np.ndarray
+            A 2D array where each row represents a transformed point in the point cloud:
+            [x, y, z, Velocity]. Returns an empty array if the sensor name is
+            not recognized.
+        """
+
+        if sensor_name not in self.sensor_config:
+            rospy.logwarn(f"Unknown sensor: {sensor_name}")
+            return np.array([])
+
+        # translation = sensor_config[sensor_name]
+        data_array = pointcloud2_to_array(msg)
+
+        # Transform data to change its frame
+        x, y, z = self.sensor_config[sensor_name]
+        # Use numpy broadcasting for better performance
+        translation = np.array([x, y, z])
+        transformed_points = np.column_stack((
+            data_array[:, :3] + translation,
+            data_array[:, 3]
+        ))
+
+        return transformed_points
     def listener(self):
```

Comment:
To fix linter error add a blank line before 'def'

```suggestion

    def listener(self):
```